testsuite/test_upload.py

```diff
@@ -0,0 +1,63 @@
+import pytest
+import base64
+import re
+from django.urls import reverse
+from hoover.search import models
+
+pytestmark = pytest.mark.django_db
+TESTFILE_PATH = '/opt/hoover/collections/testdata/test.pdf'
+
+
+def encode_filename(filename):
+    string_bytes = base64.b64encode(filename.encode('utf-8'))
+    return string_bytes.decode('utf-8')
+
+
+def setup_collection(django_user_model):
+    user = django_user_model.objects.create_user(username='testuser', password='pw')
+    col = models.Collection.objects.create(
+        name='testdata',
+        index='hoover-testdata',
+        public=True,
+        writeable=True,
+    )
+    col.users.add(user)
+    col.uploaders.add(user)
+    return col
+
+
+def test_search_upload(client, django_user_model):
+    setup_collection(django_user_model)
+    user = django_user_model.objects.get(username='testuser')
+    url = reverse('tus_upload', args=['testdata'])
+    post_headers = {
+        'HTTP_UPLOAD_METADATA': 'orig_filename ' + encode_filename('test.pdf'),
+        'HTTP_CONTENT_LENGTH': '0',
+        'HTTP_UPLOAD-LENGTH': '2681358',
+        'HTTP_TUS_RESUMABLE': '1.0.0'
+    }
+    patch_headers = {
+        'HTTP_UPLOAD_OFFSET': '0',
+        'HTTP_CONTENT_LENGTH': '1234',
+        'HTTP_CONTENT_TYPE': 'application/offset+octet-stream',
+        'HTTP_TUS_RESUMABLE': '1.0.0'
+    }
+    client.force_login(user)
+    initial_response = client.post(url, **post_headers)
+    assert initial_response.status_code == 201
+
+    upload_location = initial_response.get('Location')
+    upload_url = re.search('/api/.+', upload_location).group(0)
+
+    with open(TESTFILE_PATH, 'rb') as f:
+        data = f.read()
+
+    upload_response = client.patch(upload_url, data=data, **patch_headers)
+    assert upload_response.status_code == 204
+
+    # It seems like the "tus_upload_finished_signal" is not working properly in the tests
+    # or the handler is not being registered. So for now we are not testing, if the file is
+    # uploaded/moved to the correct location.
```

Comment:
There shouldn't be any difference in signals, so please find out why the signal isn't being sent when running tests.

You can try to mock out the actual sender https://github.com/alican/django-tus/blob/aacdaf6deca0a9f2b76a74dafa748cd4e88b667e/django_tus/views.py#L112-L120 and see if it's called, maybe it isn't. 

Hey, maybe it's [this guy's problem](https://github.com/pytest-dev/pytest-django/issues/438) -- you need to import the signal reveiver, so it actually registers

Reply:
Ah, now I see you already import the signal in the `ready` function.